scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -198,17 +198,17 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
         i
       }
       val next = if (nextPos != -1) scannerTokens(nextPos) else null
-      def nextScannerToken: Option[Token] =
-        if (currPos + 1 < scannerTokens.length) Some(scannerTokens(currPos + 1))
-        else None
       // SIP-27 Trailing comma (multi-line only) support.
       // If a comma is followed by a new line & then a closing paren, bracket or brace
       // then it is a trailing comma and is ignored.
       def isTrailingComma: Boolean =
         dialect.allowTrailingCommas &&
           curr.is[Comma] &&
           next.is[CloseDelim] &&
-          nextScannerToken.exists(_.is[LineEnd])
+          scannerTokens
+            .slice(currPos + 1, nextPos)
+            .dropWhile(t => t.is[Comment] || t.is[Space] || t.is[Tab])
+            .headOption.exists(_.is[LineEnd])
```

Comment:
It's not pretty but it does the job.

In English:

- we're on a comma
- the next non trivia is a closing delimiter (`)`, `]` or `}`)
- take all the tokens in between the comma and the closing delimiter
- drop all the tokens that are comments, spaces or tabs
- check if we have a newline

I'm open to suggestions for more elegant implementations

Reply:
This is a fairly hot path, can we make this a tailrec method instead? It's best to access tokens by index.

Reply:
Sure, that makes sense. I'll do it later